lib/linklist.h

```diff
@@ -288,6 +293,37 @@ extern void list_delete_node(struct list *list, struct listnode *node);
  */
 extern void list_add_list(struct list *list, struct list *add);
 
+/*
+ * Delete all nodes which satisfy a condition from a list.
+ * Deletes the node if cond function returns true for the node.
+ * If function ptr passed is NULL, it deletes all nodes
+ *
+ * list
+ *    list to operate on
+ * cond
+ *    function pointer which takes node data as input and return TRUE or FALSE
+ */
+
+extern void list_cond_delete_nodes(struct list *list, cond_del_fn_t  cond);
```

Comment:
I would have named this `list_filter`, as that's what this operation is called in most other programming languages, [e.g.](https://docs.python.org/3/library/functions.html#filter), [e.g.](http://hackage.haskell.org/package/base-4.12.0.0/docs/Data-List.html#v:filter). Also the styling is wrong, two spaces after the type for `cond` should be one.

Reply:
liked the suggestion list_filter. changed to list_filter_out_nodes as we will be deleting the nodes that matches the filter.